src/codegen/compile.ml

```diff
@@ -5610,33 +5637,34 @@ module Stabilization = struct
            fs
           in
           StableMem.get_mem_size env ^^
-          E.then_trap_with env "StableMem.mem_size non-zero" ^^
+          G.i (Test (Wasm.Values.I64 I64Op.Eqz)) ^^
+          E.else_trap_with env "StableMem.mem_size non-zero" ^^
```

Comment:
eqz is performing a not, followed by a conditional with only an else branch. So this looks like unnecessary double negation.

Reply:
Actually, I don't see how to improve this. `if` consumes an i32 and I've got an i64 on the stack. Is there a better way?

Reply:
Ah, you're right, I overlooked that this is an i64. Scratch my comment then.